### PR TITLE
pipeline: schedule pipelines separately

### LINKED ISSUE #1502: [BUG] byt media pipe doesn't work with "192 frames per 4000us deadline"
**Describe the bug**
Byt media pipe plays audio with funny 4-8 times speed if you set up 
"192 frames per 4000us deadline" in topology. However,  "48 frames per 1000us deadline" works.

**To Reproduce**
Just play audio to media pipe hw:0,1

**Reproduction Rate**
Always

**Expected behavior**
Normal speed

**Impact**
Can't change scheduling params

**Environment**
1) kernel: latest, fw:latest
2) topology: sof-byt-rt5682.tplg
3) platform: byt
4) Reproducibility Rate. Always

### PR DESCRIPTION
**Pull request by tlauda**
This patch enables separate scheduling of connected
pipelines. It allows for different scheduling periods
between connected pipelines and flexible controlling
of such topologies.

Signed-off-by: Tomasz Lauda <tomasz.lauda@linux.intel.com>

Fixes #1502.

### CODE REVIEW

File: src/include/sof/schedule/task.h [hunk 1]
```diff
@@ -53,6 +55,15 @@ struct task {
 	void *private;
 };
 
+/** \brief Task type registered by pipelines. */
+struct pipeline_task {
```
Comment (lgirdwood) [inline src/include/sof/schedule/task.h]: Why do we need to have a new task type. What is regular struct task missing ?

Comment (tlauda) [reply src/include/sof/schedule/task.h]: It holds additional information, which are not needed to be included in generic task structure. Think of it like class inheritance.

Comment (lgirdwood) [reply src/include/sof/schedule/task.h]: ok, I assume we are not changing any of our schedule_ APIs to use this struct ? i.e. they will still use task. This structure will only be used by pipeline ?

Btw, "registrable" seems ambiguous. Does it mean we are scheduling on IRQ ?

Comment (tlauda) [reply src/include/sof/schedule/task.h]: API is not changed.

**Pull request merged**